src/main/java/apoc/export/util/FileUtils.java

```diff
@@ -23,6 +23,21 @@ public static CountingReader readerFor(String fileName) throws IOException {
         return new CountingReader(file);
     }
 
+    public static String changeFileUrlIfImportDirectoryConstrained(String url) throws IOException {
+        if (isFile(url) && ApocConfiguration.isEnabled("import.file.use_neo4j_config")) {
+            if (!ApocConfiguration.isEnabled("import.file.allow_read_from_filesystem"))
+                throw new RuntimeException("Import file "+url+" not enabled, please set dbms.security.allow_csv_import_from_file_urls=true in your neo4j.conf");
+            String importDir = ApocConfiguration.get("import.file.directory", null);
+            if (importDir != null && !importDir.isEmpty()) {
+                File dir = new File(importDir);
+                if (dir.exists() && dir.isDirectory() && dir.canRead()) {
+                    return "file:" + new File(importDir, url.substring(url.indexOf("/")+1)).getAbsolutePath();
+                }
+            }
+        }
+        return url;
+    }
+
     public static boolean isFile(String fileName) {
         if (fileName==null) return false;
         if (fileName.toLowerCase().startsWith("http")) return false;
```

Comment:
we should fix this to work with URLs in general
i.e. `return !isUrl(fileName) || new URI(fileName).getProtocol().equalsIgnoreCase("file");`